regolith/helpers/makeappointmentshelper.py

```diff
@@ -0,0 +1,194 @@
+                            day_burn = total_burn[x].get('postdoc_days')
+                        elif appt.get('type') == 'ss':
+                            day_burn = total_burn[x].get('ss_days')
+                        if day_burn < 0:
+                            depleted.append("    person: {}, appointment: {}, grant: {},\n"
+                                            "            from {} until {}".format(
+                                person.get('_id'), appt.get('_id'), grant.get('_id'), str(day), str(appt_end)))
+                            depleted_period = True
+
+        grants_begin = min(get_dates(grant)['begin_date'] for grant in self.gtx["grants"])
+        grants_end = max(get_dates(grant)['end_date'] for grant in self.gtx["grants"])
+        grants_timespan = grants_end - grants_begin
+        datearray = []
+
+        for x in range(grants_timespan.days + 1):
+            datearray.append(grants_begin + relativedelta(days=x))
+        dates = matplotlib.dates.date2num(datearray)
+        cum_student, cum_pd, cum_ss = [0.0] * len(datearray), [0.0] * len(datearray), [0.0] * len(datearray)
+
+        for grant in self.gtx["grants"]:
+            if grant.get('_id') in BLACKLIST or grant.get('alias') in BLACKLIST:
+                continue
+            grt_begin, grt_end = get_dates(grant)['begin_date'], get_dates(grant)['end_date']
+            grt_duration = (grt_end - grt_begin).days + 1
+            grant_dates = []
+            this_student, this_pd, this_ss = [0.0] * grt_duration, [0.0] * grt_duration, [0.0] * grt_duration
+            grant_amts = grant_burn(grant, all_appts, begin_date=grt_begin, end_date=grt_end)
+            grant_amt = grant_amts[-1].get('student_days') + grant_amts[-1].get('postdoc_days') + grant_amts[-1].get('ss_days')
+            if grant_amt > 30.5:
+                underspent.append("    {}: grant: {}, underspend amount: {} months".format(
+                    str(grt_end), grant.get('_id'), round(grant_amt/30.5, 2)))
+            elif grant_amt < -30.5:
+                overspent.append("    {}: grant: {}, overspend amount: {} months".format(
+                    str(grt_end), grant.get('_id'), round(grant_amt/30.5, 2)))
+            counter = 0
+            for x in range (grants_timespan.days + 1):
+                if is_current(grant, now=datearray[x]):
+                    grant_dates.append(datearray[x])
+                    this_student[counter] = grant_amts[counter].get('student_days')
+                    cum_student[x] += grant_amts[counter].get('student_days')
+                    this_pd[counter] = grant_amts[counter].get('postdoc_days')
+                    cum_pd[x] += grant_amts[counter].get('postdoc_days')
+                    this_ss[counter] = grant_amts[counter].get('ss_days')
+                    cum_ss[x] += grant_amts[counter].get('ss_days')
+                    counter += 1
+            plt.plot_date(grant_dates, this_student, ls='-', marker="", label="student days")
```

Comment:
it is probably a good idea to pull all the plotting logic out into a function.  It won't be a very reusable function so leave it here in this module but maybe outside the helper class.  I am not sure how easily it can be tested, but if you can test it, put the tests into ``test_helpers.py``.

This will make it easier to maintain and turn it on and off if needs be.

Reply:
done adding the function! testing might take a bit more time, however. also, would you like me to have a 
`subpi.add_argument("-p", "--plot", action="store_true", help='plot grant burns')
`
in the subparser?